x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -108,31 +112,64 @@ export const generateMlInferencePipelineBody = ({
 
 export const getSetProcessorForInferenceType = (
   destinationField: string,
-  inferenceType: string
+  inferenceType: string,
+  pipelineName: string
 ): IngestSetProcessor | undefined => {
   let set: IngestSetProcessor | undefined;
   const prefixedDestinationField = `ml.inference.${destinationField}`;
+  const onFailure: IngestProcessorContainer[] = [
+    {
+      append: {
+        field: '_source._ingest.set_errors',
+        ignore_failure: true,
+        value: [
+          {
+            message: `Processor 'set' in pipeline '${pipelineName}' failed with message '{{ _ingest.on_failure_message }}'`,
+            pipeline: pipelineName,
+            timestamp: '{{{ _ingest.timestamp }}}',
+          },
+        ],
+      },
+    },
+  ];
 
   if (inferenceType === SUPPORTED_PYTORCH_TASKS.TEXT_CLASSIFICATION) {
     set = {
       copy_from: `${prefixedDestinationField}.predicted_value`,
       description: `Copy the predicted_value to '${destinationField}' if the prediction_probability is greater than 0.5`,
       field: destinationField,
       if: `ctx.${prefixedDestinationField}.prediction_probability > 0.5`,
```

Comment:
This is where the failure occurs if there is a hyphen/dash in `prefixedDestinationField`. We can fix this by changing this to use square brackets and single quotes like this:
```suggestion
      if: `ctx.ml.inference.['${destinationField}'].prediction_probability > 0.5`,
```
But as [noted in Slack](https://elastic.slack.com/archives/CHY7020LT/p1668468641993699), there could be an injection vulnerability here. And in that case, the square brackets do not help.

Reply:
We reached an agreement that this is not an issue.